src/WebJobs.Extensions.OpenAI.AISearch/AzureAISearchProvider.cs

```diff
@@ -0,0 +1,323 @@
+            VectorSearch = new()
+            {
+                Algorithms =
+                {
+                    new HnswAlgorithmConfiguration(vectorSearchConfigName)
+                },
+                Profiles =
+                {
+                    new VectorSearchProfile(vectorSearchProfile, vectorSearchConfigName)
+                }
+            },
+            Fields =
+            {
+                new SimpleField("id", SearchFieldDataType.String) { IsKey = true },
+                new SearchableField("text") { AnalyzerName = LexicalAnalyzerName.EnMicrosoft },
+                new SimpleField("title", SearchFieldDataType.String) { IsFacetable = true },
+                new SearchField("embeddings", SearchFieldDataType.Collection(SearchFieldDataType.Single))
+                {
+                    VectorSearchDimensions = this.VectorSearchDimensions,
+                    IsSearchable = true,
+                    VectorSearchProfileName = vectorSearchProfile,
+                },
+                new SimpleField("timestamp", SearchFieldDataType.DateTimeOffset) { IsFacetable = true }
+            },
+            SemanticSearch = new()
+            {
+                Configurations =
+                {
+                    new SemanticConfiguration("default", new()
+                    {
+                        ContentFields =
+                        {
+                            new SemanticField("text")
+                        },
+                        TitleField = new SemanticField("title")
+                    })
+                }
+            }
+        };
+
+        await searchIndexClient.CreateIndexAsync(index, cancellationToken);
+    }
+
+    async Task IndexSectionsAsync(SearchClient searchClient, SearchableDocument document, CancellationToken cancellationToken = default)
+    {
+        int iteration = 0;
+        IndexDocumentsBatch<SearchDocument> batch = new();
+        for (int i = 0; i < document.Embeddings.Response.Data.Count; i++)
+        {
+            batch.Actions.Add(new IndexDocumentsAction<SearchDocument>(
+                IndexActionType.MergeOrUpload,
+                new SearchDocument
+                {
+                    ["id"] = Guid.NewGuid().ToString("N"),
+                    ["text"] = document.Embeddings.Request.Input![i],
+                    ["title"] = Path.GetFileNameWithoutExtension(document.Title),
+                    ["embeddings"] = document.Embeddings.Response.Data[i].Embedding.ToArray() ?? Array.Empty<float>(),
+                    ["timestamp"] = DateTime.UtcNow
+                }));
+            iteration++;
+            if (iteration % 1_000 is 0)
+            {
+                // Every one thousand documents, batch create.
+                IndexDocumentsResult result = await searchClient.IndexDocumentsAsync(batch, cancellationToken: cancellationToken);
+                int succeeded = result.Results.Count(r => r.Succeeded);
+                this.logger.LogDebug("""
```

Comment:
I wonder if `LogInformation` would be more appropriate for these two log statements.

Reply:
updated to LogInformation